instrumentation/classloading/src/main/java/io/opentelemetry/auto/instrumentation/classloading/ClassloadingInstrumentation.java

```diff
@@ -0,0 +1,78 @@
+package io.opentelemetry.auto.instrumentation.classloading;
+
+import static io.opentelemetry.auto.tooling.ByteBuddyElementMatchers.safeHasSuperType;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isProtected;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.isStatic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.auto.tooling.Constants;
+import io.opentelemetry.auto.tooling.Instrumenter;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(Instrumenter.class)
+public final class ClassloadingInstrumentation extends Instrumenter.Default {
+  public ClassloadingInstrumentation() {
+    super("classloading");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return safeHasSuperType(named("java.lang.ClassLoader"));
```

Comment:
Worth excluding the internal classloader or any classloader that we know delegates properly?

Reply:
ya, makes sense to exclude `java.lang.ClassLoader` itself. not sure any others are really worth it, wdyt?

Reply:
I think at least `AgentClassLoader` too.  I guess any other classloader could either explicitly or implicitly not delegate, so I think those are the only two that could work.

Reply:
done